### PR TITLE
Added spec for tendrl-gdeploy wrapper

### PR DESCRIPTION
**Pull request by nnDarshan**
tendrl-bug-id: Tendrl/specifications#49
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@brainfunked @sac @shtripat @nthomas-redhat Please review

File: specs/tendrl_gdeploy_wrapper.adoc [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+== Use Cases
+
+* tendrl-gdeploy wrapper can be used for provisioning gluster nodes
+* It can be used for gluster performing volume management operations
+
+== Proposed change
+=== gdeploy
+* Move the subscription management and repository setup playbook out of gdeploy
+  and make available as a standalone project that both tendrl and gdeploy can use.
+* Confirm the remove node functionality with gluster folk.
```
Comment (shtripat) [inline specs/tendrl_gdeploy_wrapper.adoc]: Looks bit adhoc comment :)

File: specs/tendrl_gdeploy_wrapper.adoc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+This wrapper should have a mechanism to generate the gdeploy config files for the
+tasks that tendrl intends to perform via gdeploy.
+
+== Use Cases
+
+* tendrl-gdeploy wrapper can be used for provisioning gluster nodes
+* It can be used for gluster performing volume management operations
+
+== Proposed change
+=== gdeploy
```
Comment (mkudlej) [inline specs/tendrl_gdeploy_wrapper.adoc:26]: Are there any BZs or upstream GDeploy tickets for tracking these changes, please?

File: specs/tendrl_gdeploy_wrapper.adoc [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+None
+
+=== Performance Impact
+
+None
+
+=== Other deployer impact
+
+Now one of the gluster nodes has to be picked as provisoner and that node
+has to accept all the gluster provisioning related tasks.
```
Comment (r0h4n) [inline specs/tendrl_gdeploy_wrapper.adoc:134]: This requires changes in the node-agent, have you listed them ?

File: specs/tendrl_gdeploy_wrapper.adoc [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+Other contributors:
+None
+
+=== Work Items
+
+* Write a gdeploy config generator
+
+* Write gdeploy executor, which accepts config file mentioned in previous
+  step and wait for the results and update operation logs accordingly
```
Comment (r0h4n) [inline specs/tendrl_gdeploy_wrapper.adoc:158]: how is gdeploy reporting logs back to tendrl?

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: specs/tendrl_gdeploy_wrapper.adoc (hunk 1 above)
Comment (nnDarshan) [reply specs/tendrl_gdeploy_wrapper.adoc]: Changed this

File: specs/tendrl_gdeploy_wrapper.adoc (hunk 3 above)
Comment (nnDarshan) [reply specs/tendrl_gdeploy_wrapper.adoc:134]: will add this detail under work-item section

File: specs/tendrl_gdeploy_wrapper.adoc (hunk 4 above)
Comment (nnDarshan) [reply specs/tendrl_gdeploy_wrapper.adoc:158]: Current gdeploy dumps the output that is returned by ansible, which is not of help to tendrl. However gdeploy team is working to improve the output of gdeploy , this change is not available in master branch of gdeploy project. There is a wip branch that has this feature.

**Pull request merged**